aptos-move/framework/aptos-framework/sources/stake.spec.move

```diff
@@ -72,6 +105,108 @@ spec aptos_framework::stake {
         ensures exists<ValidatorPerformance>(aptos_addr);
     }
 
+    spec join_validator_set(
+        operator: &signer,
+        pool_address: address
+    )
+    {
+        // This function casue timeout (property proved)
+        pragma verify_duration_estimate = 120;
+        aborts_if !staking_config::get_allow_validator_set_change(staking_config::get());
+        aborts_if !exists<StakePool>(pool_address);
+        aborts_if !exists<ValidatorConfig>(pool_address);
+        aborts_if !exists<ValidatorSet>(@aptos_framework);
+
+        let stake_pool = global<StakePool>(pool_address);
+        let validator_set = global<ValidatorSet>(@aptos_framework);
+        let post p_validator_set = global<ValidatorSet>(@aptos_framework);
+        aborts_if signer::address_of(operator) != stake_pool.operator_address;
+        aborts_if option::spec_is_some(spec_find_validator(validator_set.active_validators, pool_address)) ||
+                    option::spec_is_some(spec_find_validator(validator_set.pending_inactive, pool_address)) ||
+                        option::spec_is_some(spec_find_validator(validator_set.pending_active, pool_address));
+
+        let config = staking_config::get();
+        let voting_power = get_next_epoch_voting_power(stake_pool);
+
+        let minimum_stake = config.minimum_stake;
+        let maximum_stake = config.maximum_stake;
+        aborts_if voting_power < minimum_stake;
+        aborts_if voting_power >maximum_stake;
+
+        let validator_config = global<ValidatorConfig>(pool_address);
+        aborts_if vector::is_empty(validator_config.consensus_pubkey);
+
+        let validator_set_size = vector::length(validator_set.active_validators) + vector::length(validator_set.pending_active);
```

Comment:
Since `validator_set.pending_active` is updated here: `vector::push_back(&mut validator_set.pending_active, generate_validator_info(pool_address, stake_pool, *validator_config)); `, `validator_set_size` should reflect this?